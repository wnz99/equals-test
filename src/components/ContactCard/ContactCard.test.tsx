import { render, screen, waitFor } from '@testing-library/react'
import React from 'react'

import { ContactCard } from './index'

describe('<ContactCard />', () => {
  it('should render without errors', async () => {
    const contact = {
      createdAt: '2023-02-03T03:53:58.977Z',
      name: 'Sassy Michelle',
      avatar:
        'https://www.meme-arsenal.com/memes/3ee2c16ed2c4be36c7b10c89ed27ad13.jpg',
      email: 'sassy@email.com',
      phone: '23456789',
      birthday: '15/11/1963',
      id: '130',
    }

    const { asFragment } = render(<ContactCard contact={contact} />)

    expect(asFragment()).toMatchSnapshot()
  })

  it('should show View button', async () => {
    const contact = {
      createdAt: '2023-02-03T03:53:58.977Z',
      name: 'Sassy Michelle',
      avatar:
        'https://www.meme-arsenal.com/memes/3ee2c16ed2c4be36c7b10c89ed27ad13.jpg',
      email: 'sassy@email.com',
      phone: '23456789',
      birthday: '15/11/1963',
      id: '130',
    }

    render(<ContactCard contact={contact} />)

    await waitFor(async () => {
      const viewButton = screen.getByText(/View/i)

      expect(viewButton).toBeInTheDocument()

      expect(viewButton.closest('a')).toHaveAttribute('href', '/contacts/130')
    })
  })
})
