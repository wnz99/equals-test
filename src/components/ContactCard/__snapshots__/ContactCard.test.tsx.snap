// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactCard /> should render without errors 1`] = `
<DocumentFragment>
  <div
    class="chakra-card css-3cb9se"
  >
    <div
      class="chakra-card__body css-0"
    >
      <div
        class="css-gmuwbf"
      >
        <img
          alt="Avatar of Sassy Michelle"
          class="chakra-image css-j17amb"
          src="https://www.meme-arsenal.com/memes/3ee2c16ed2c4be36c7b10c89ed27ad13.jpg"
        />
      </div>
      <div
        class="chakra-stack css-10aolb9"
      >
        <h2
          class="chakra-heading css-0"
        >
          Sassy Michelle
        </h2>
      </div>
    </div>
    <div
      class="chakra-card__footer css-k008qs"
    >
      <div
        class="chakra-button__group css-1mlmbpp"
        data-orientation="horizontal"
        role="group"
      >
        <a
          href="/contacts/130"
        >
          <button
            class="chakra-button css-1fysbuj"
            type="button"
          >
            View
          </button>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
